### ISSUE TITLE
Pretraining in benchmark.py

### ISSUE THREAD

**Issue by erinaldiq** [CQCL/qbm_benchmark_dataset#15]
We want to run multiple simulations of the QBM training using the benchmark.py script. 

Currently the script does not allow us to run pretraining as discussed in the notebook. 

This is how I think we can add pretraining to the script:
1. We need 2 "models": right now we select one label for the model Hamiltonian but we should be able to pass 2 different labels
1b. The two Hamiltonians should have this constraint: the terms of the pretraining models are included in the set of terms of the full model
2. We need 2 training loops, so we have to allow for different number of epochs and different learning rates. 
2b. After the first training round, we need to initialize the full model from the set of trained parameters coming from the first round. 
3. We can implement the above changes by allowing for some input arguments to be lists of 2 elements, where the first element of the list refers to the pretraining (first) round.
